src/main/java/org/spongepowered/api/data/key/Keys.java

```diff
@@ -2097,6 +2097,14 @@
      */
     public static final Key<OptionalValue<UUID>> TAMED_OWNER = DummyObjectProvider.createExtendedFor(Key.class,"TAMED_OWNER");
 
+    /**
+     * Represents the {@link Key} for a targeted entity,
+     * like by a {@link ShulkerBullet}.
+     *
+     * @see TargetedEntityData#value
+     */
+    public static final Key<Value<Entity>> TARGETED_ENTITY = DummyObjectProvider.createExtendedFor(Key.class,"TARGETED_ENTITY");
```

Comment:
I'm not a fan of reintroducing data targeting live `Entity` instances, Data API doesn't do well with it since it's not an immutable variation. When we handled targeting for entity AI, we elected to not represent that via Data API. I'm not against it, but I'm not a fan.